docs/import/add_disaster.js

```diff
@@ -0,0 +1,220 @@
+import {eeLegacyPathPrefix, gdEePathPrefix} from '../ee_paths.js';
+import {getFirestoreRoot} from '../firestore_document.js';
+
+export {
+  enableWhenReady,
+};
+// Visible for testing
+export {
+  addDisaster,
+  createOptionFrom,
+  createStateAssetPickers,
+  disasters,
+  emptyCallback,
+  writeDisaster,
+};
+
+// Currently a map of disaster name to states. This pulls once on firebase
+// authentication and then makes local updates afterwards so we don't need to
+// wait on firebase writes to read new info.
+const disasters = new Map();
+
+const SENTINEL_OPTION_VALUE = '...';
+const SENTINEL_NEW_DISASTER_VALUE = 'NEW DISASTER';
+
+/**
+ * Populates the disaster picker with disasters from firestore, and enables
+ * the ability to add a new disaster.
+ * @return {Promise<firebase.firestore.QuerySnapshot>}
+ */
+function enableWhenReady() {
+  // enable (currently hidden) add disaster button now that firestore is ready.
+  const addDisasterButton = $('#add-disaster-button');
+  addDisasterButton.prop('disabled', false);
+  addDisasterButton.on('click', addDisaster);
+
+  // populate disaster picker.
+  const disasterPicker = $('#disaster');
+  disasterPicker.append(createOptionFrom(SENTINEL_OPTION_VALUE));
+  return getFirestoreRoot()
+      .collection('disaster-metadata')
+      .get()
+      .then((querySnapshot) => {
+        querySnapshot.forEach((doc) => {
+          const name = doc.id;
+          disasterPicker.append(createOptionFrom(name));
+          disasters.set(name, doc.data().states);
+        });
+        disasterPicker.append(
+            createOption('ADD NEW DISASTER', SENTINEL_NEW_DISASTER_VALUE));
+        disasterPicker.on('change', () => toggleState($('#disaster').val()));
+      });
+}
+
+/**
+ * Switch between the page creating a new disaster and the page editing a
+ * current disaster.
+ * @param {String} disaster
+ * @return {Promise<void>} completes when we've finished filling all state
+ * pickers.
+ */
+function toggleState(disaster) {
+  if (disaster === SENTINEL_NEW_DISASTER_VALUE) {
+    $('#new-disaster').show();
+    $('#selected-disaster').hide();
+  } else {
+    // TODO: display more disaster info including current layers etc. We
+    // might not want want to create these asset pickers every time we toggle
+    // between disasters because it requires ee calls. For now it's quite fast,
+    // will handle in follow up PRs if it begins to be problem.
+    $('#new-disaster').hide();
+    $('#selected-disaster').show();
+    return createStateAssetPickers(disasters.get(disaster));
+  }
+}
+
+/**
+ * Writes the given details to a new disaster entry in firestore. Fails if
+ * there is an existing disaster with the same details.
+ * @param {string} disasterId of the form <year>-<name>
+ * @param {Array<string>} states array of state (abbreviations)
+ * @return {Promise<boolean>} returns true after successful write to firestore.
+ */
+function writeDisaster(disasterId, states) {
```

Comment:
nit: maybe writeNewDisaster or addDisaster to disambiguate from modifying an existing disaster?